xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-ui/src/main/resources/XWiki/SearchSuggestCode.xml

```diff
@@ -0,0 +1,101 @@
+  <version>1.1</version>
+  <title>SearchSuggestCode</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+## Iterate over the sources defined in the configuration document, and create a source array to be passed to the
+## search suggest contructor.
+#set ($sources = [])
+#set ($searchSuggestConfig = $xwiki.getDocument('XWiki.SearchSuggestConfig'))
+#foreach ($source in $searchSuggestConfig.getObjects('XWiki.SearchSuggestSourceClass'))
+  #if ($source.getProperty('activated').value == 1)
+    #set ($engine = $source.getProperty('engine').value)
+    #if ("$!engine" == '')
+      ## For backward compatibility we consider the search engine to be Lucene when it's no specified.
+      #set ($engine = 'lucene')
+    #end
+    #set ($discard = $xwiki.getDocument('XWiki.SearchCode').getRenderedContent())
+    #if ($engine == $searchEngine)
+      #try("evaluateException")
+        #set ($evaluatedSource = $source.evaluate())
+      #end
+      #if ("$!evaluateException" != '')
+        #set ($discard = $logtool.error(
+          "Error when trying to evaluate XWiki.SearchSuggestSourceClass: $evaluateException"))
+        #set ($discard = $logtool.debug($exceptiontool.getStackTrace($evaluateException)))
+      #end
+      #set ($name = $source.getProperty('name').value)
+      #if ($services.localization.get($name))
+        #set ($name = $services.localization.render($name))
+      #elseif ("$!evaluatedSource" != '')
+        ## Evaluate the Velocity code for backward compatibility.
+        #set ($name =  $evaluatedSource.name)
+      #end
+      #set ($icon = $source.getProperty('icon').value)
+      #if ($icon.startsWith('icon:'))
+        #set ($icon = $xwiki.getSkinFile("icons/silk/${icon.substring(5)}.png"))
+      #elseif ("$!evaluatedSource" != '')
+        ## Evaluate the Velocity code for backward compatibility.
+        #set ($icon = $evaluatedSource.icon)
+      #end
+      #set ($service = $source.getProperty('url').value)
+      #set ($parameters = {
+        'query': $source.getProperty('query').value,
+        'nb': $source.getProperty('resultsNumber').value
+      })
+      #if ($xwiki.exists($service))
+        #set ($discard = $parameters.put('outputSyntax', 'plain'))
+        #set ($service = $xwiki.getURL($service, 'get', $escapetool.url($parameters)))
+      #else
+        ## Evaluate the Velocity code for backward compatibility.
+        #set ($service = "#evaluate($service)")
+        #set ($service = "$service#if ($service.contains('?'))&amp;#else?#end$escapetool.url($parameters)")
+      #end
+      #set ($highlight = $source.getProperty('highlight').value == 1)
+      #set ($discard = $sources.add({
+        'name': $name,
+        'varname': 'input',
+        'script': $service,
+        'icon': $icon,
+        'highlight': $highlight
+      }))
+    #end
+  #end
+#end
+$jsontool.serialize($sources)
```

Comment:
That's (potentially) a security vulnerability. Please use `#jsonResponse` and never, ever output the result of `$jsontool.serialize` in a context where XWiki syntax is enabled.